src/android/EncryptionExecutor.java

```diff
@@ -0,0 +1,39 @@
+package com.crypho.plugins;
+
+import android.content.Context;
+import android.util.Base64;
+import android.util.Log;
+import org.json.JSONException;
+import org.json.JSONObject;
+import java.util.concurrent.Callable;
+
+public class EncryptionExecutor implements Callable<String>{
+
+    private static final String TAG = "SecureStorage";
+    private String service, key, value;
+    private Context context;
+
+    public EncryptionExecutor(String service, String key, String value, Context context){
+        this.service = service;
+        this.key = key;
+        this.value = value;
+        this.context = context;
+    }
+    @Override
+    public String call() throws JSONException {
+        try {
+            JSONObject result = AES.encrypt(value.getBytes(), service.getBytes());
+            byte[] aes_key = Base64.decode(result.getString("key"), Base64.DEFAULT);
+            byte[] aes_key_enc = RSA.encrypt(aes_key, service2alias(service));
+            result.put("key", Base64.encodeToString(aes_key_enc, Base64.DEFAULT));
+            return result.toString();
+        } catch (Exception e) {
+            Log.e(TAG, "Encrypt (RSA/AES) failed :", e);
+            return new JSONObject().put("ERROR",e.getMessage()).toString();
```

Comment:
Same deal as the DecryptionExecutor. Return the whole JSONObject or use a new POJO.